### PR TITLE
Add Argument::in() and Argument::notIn()

### PR DESCRIPTION
**Pull request by viniciusalonso**
I was working in some exercises using TDD and I needed to write an example like this:

```php
$rule = $this->prophesize(MultipleThreeRule::class);
                                                                        
$rule->condition(Argument::that(function($arg) {
    return in_array($arg, [3, 6, 9, 12]);
}))->willReturn(true);
```
The idea is guarantee that a value is in array.

```php
$rule->condition(12); // returns true
``` 
Then, I implemented an easier way to get to same result:

```php
$rule = $this->prophesize(MultipleThreeRule::class);
                                                                        
$rule->condition(Argument::in([3, 6, 9, 12]))->willReturn(true);
```

And I also I implemented the opposite way, when value is not in array:

```php
$rule = $this->prophesize(MultipleThreeRule::class);
                                                                        
$rule->condition(Argument::notIn([3, 6, 9, 12]))->willReturn(false);
```

### CODE REVIEW

File: src/Prophecy/Argument/Token/InArrayToken.php [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+ * @author Vinícius Alonso <vba321@hotmail.com>
+ */
+class InArrayToken implements TokenInterface
+{
+    private $token = array();
+
+    /**
+     * @param array $arguments tokens
+     */
+    public function __construct(array $arguments)
```
Comment (Taluu) [inline src/Prophecy/Argument/Token/InArrayToken.php]: you could probably add a `$strict` argument (defaulting to `true` ?) so we can have an optimized call on `in_array`

Comment (viniciusalonso) [reply src/Prophecy/Argument/Token/InArrayToken.php]: It's interesting, what should be exactly the `$strict` role in that method? I'd like to understand better your idea.

Comment (Taluu) [reply src/Prophecy/Argument/Token/InArrayToken.php]: `in_array` have a 3rd parameter for that. If it's `true` and a `use function in_array` or `\in_array` is used, an optimized opcode will be used.

Comment (viniciusalonso) [reply src/Prophecy/Argument/Token/InArrayToken.php]: Good point! Let me improve this chunk of code, thanks. :)

**Comment by ciaranmcnulty** [comment]
Thanks @viniciusalonso - looks good

**Pull request merged**